core/modules/node/node.types.inc

```diff
@@ -299,6 +299,23 @@ function node_type_form($form, &$form_state, $type = NULL) {
     '#title' => t('Create new revision by default'),
     '#default_value' => $type->settings['revision_default'],
     '#parents' => array('revision_default'),
+    '#states' => array(
+      'visible' => array(
+        'input[name="revision_enabled"]' => array('checked' => TRUE),
+      ),
+    ),
+  );
+  $form['revision']['revision']['revision_required'] = array(
```

Comment:
@BWPanda can we please remove the extraneous `['revision']` from this checkbox and the 2 checkboxes before that? It seems that they have been added accidentally.

Thanks.